### PR TITLE
Update authentication.md

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> For 1.8 Features: set Milestone to `1.8` and Base Branch to `release-1.8`
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>
> NOTE: Please check the “Allow edits from maintainers” box (see image below) to
> [allow reviewers to fix problems](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) on your patch and speed up the review process.
>
> Please delete this note before submitting the pull request.

Allow edits from maintainers checkbox

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5385)

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@tengqm I think it is confused, client should be authenticated passively

**Comment by chenopis** [comment]
Is the process of authentication part of the request? Or do things have to already be authenticated before they can make a request?

**Comment by hzxuzhonghu** [comment]
@chenopis yeal, I think authentication is a must, in filter before real handler

**Comment by tengqm** [comment]
@hzxuzhonghu I meant that the grammar of the text doesn't have to be fixed.

**Comment by chenopis** [comment]
@tengqm I think that if authentication is a prerequisite before the API request can be made, then I think "must be authenticated" is appropriate. Otherwise, it is implying that the authentication happens as part of the request.

**Comment by tengqm** [comment]
@chenopis My point was that there is no need to change the text because 
"must authenticate when making requests to the API server" is the same as the revised text.
It is okay to say "a user must authenticate to the API server". It doesn't have to be "a user must be authenticated to the API server". That is ... there is no language issue here IMHO.

**Comment by tengqm** [comment]
@chenopis Previous response was purely from language perspective. From technical perspective, this revision is also not improving anything. A request, when arrives at the API server, will go through a pipeline of middleware (aka. plugins): PanicyRecovery, RequestContext, RequestInfo, TimeoutForNonLongRunningRequest, CORS etc before it reaches the Authentication phase. After Authentication, there are Impersonation, MaxInFlightLimit and Authorization plugins.
In other words, the authentication happens *when* a request is being processed, *not* before it.
